go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -193,51 +200,117 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 		return errs.Error()
 	}
 
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-
-		// Update the master field in the global shard record. We don't use a lock
-		// here anymore. The lock was only to ensure that the global shard record
-		// didn't get modified between the time when we read it and the time when we
-		// write it back. Now we use an update loop pattern to do that instead.
-		event.DispatchUpdate(ev, "updating global shard record")
-		log.Infof("finalizeTabletExternallyReparented: updating global shard record if needed")
-		_, err = agent.TopoServer.UpdateShardFields(ctx, tablet.Keyspace, tablet.Shard, func(currentSi *topo.ShardInfo) error {
-			if topoproto.TabletAliasEqual(currentSi.MasterAlias, tablet.Alias) {
-				return topo.NewError(topo.NoUpdateNeeded, tablet.Alias.String())
-			}
-			if !topoproto.TabletAliasEqual(currentSi.MasterAlias, oldMasterAlias) {
-				log.Warningf("old master alias (%v) not found in the global Shard record i.e. it has changed in the meantime."+
-					" We're not overwriting the value with the new master (%v) because the current value is probably newer."+
-					" (initial Shard record = %#v, current Shard record = %#v)",
-					oldMasterAlias, tablet.Alias, si, currentSi)
-				return topo.NewError(topo.NoUpdateNeeded, oldMasterAlias.String())
-			}
-			currentSi.MasterAlias = tablet.Alias
-			return nil
-		})
-		if err != nil {
-			errs.RecordError(err)
+	masterTablet := agent.Tablet()
+
+	// Update the master field in the global shard record. We don't use a lock
+	// here anymore. The lock was only to ensure that the global shard record
+	// didn't get modified between the time when we read it and the time when we
+	// write it back. Now we use an update loop pattern to do that instead.
+
+	// We also used to do this in parallel with RefreshState on the old master
+	// we don't do that any more. we want to update the shard record first
+	// and only then attempt to refresh the old master because it is possible
```

Comment:
Actually, now that you spell out the reasoning, I don't think we need to change this. The old master being unreachable won't stop the shard record update if they're happening in parallel.

I misspoke earlier when I said, "After updating the shard record, [...] refresh the old master". I don't actually know of a reason we need to change these from parallel to series; I just forgot they were already in parallel and worded it as if they were in series. Did you think of a reason it needs to be in series?